### DISCUSSION TITLE
How to update without serial connection the Sonoff T1 (from 7.x  or 8.x to 10.x)

### DISCUSSION THREAD

**Discussion by trex2000** [arendst/Tasmota#14751] [Q&A] [answered]
Today I’ve updated to Home Assistant 2022.2.2.
I quickly noticed that none of the Sonoff T1,T2 wall swiches (ESP8285) no longer worked.
They were connected via MQTT, SetOption19 1.

After searching through the forums, I came across this:
The MQTT lights no longer support, the previously deprecated, value_template;state_value_template should be used instead.
and the solution was to flash Tasmota 10.x

The problem is that these devices cannot be updated via OTA, as there is not enough free space to install it.
Then, I’ve tried with one of the 15+ devices and flashed the minimal bin, which was flashed with success, but after flashing the normal tasmota.bin the device went silent. Web interface no longer available, does not connect to the network and won’t create an AP.

So the only solution is to remove it from the wall and flash it via serial.
Doing this for the almost 20 devices would be very time consuming.

Do you have experience with a method that work on these switches ? (Flashing certain versions and incrementally upgrade? ) I do not want to try out with the other devices, and I thought first I would ask for help here.

**Comment by barbudor** [comment, marked as answer]
There is always enough free space, you just don't follow the instructions.

Follow exactly the documentation for [Upgrading](https://tasmota.github.io/docs/Upgrading/) especially always use the "lite" version until you reach v10.1.0. **Never use tasmota-minimal** for this upgrade path.
The 1st paragraph of the  "[Upgrade flow](https://tasmota.github.io/docs/Upgrading/#upgrade-flow)" provide direct links to the appropriate binaries. Use those and no others for upload. For > 8.5.1, upload the GZ file, do not attempt to unzip it.

At the **latest step**, upgrade from tasmota-lite to full tasmota to get all features.
For this, it is always easier to use the OTA directly from the device using `http://ota.tasmota.com/tasmota/release/tasmota.bin.gz`

If you are unsure, please ask before bricking your device.
I would suggest that you test 1st with a device on table before attempting to upgrade unreachable units.
Note that following thee upgrade path should be safe, however some times it fails and serial will be your only way.

**Reply by trex2000** [reply]
So, when I first tried,  I went on flashing the minimal bin, becase I got the error: Upload buffer miscompare" error..
So, this mean that from whatever full tasmota version I have (7.x or 8.x) I can directly flash the next "lite" tasmota from the tutorial, and go incrementally up until the latest lite version, and from there I can flash the normal one, without using minimal at all ?
Or first should I flash minimal and go from there with the Lite versions ?

One more question: Why did the flashing "brick" the device when flashing from minimal to full bin ? Is there no estimated flash size check implemented in this release ? I would have expected to fail and stay with the minimal bint at least..

**Reply by trex2000** [reply]
And also, this means that no OTA update is possible for the Sonoff T1, right ? Always you have to flash the tasmota lite and then the tasmota.bin manually (or , now the gzipped archive, from 8.5.1) ?

**Reply by barbudor** [reply]
> https://tasmota.github.io/docs/Commands/#serial-bridge

Yes, "lite" is small enough to be flashed without using minimal and ncludes everything that is necessary to support the upgrade

> from there I can flash the normal one, without using minimal at all ?

Yes, you should be able to go from a "lite" to a "full" without minimal
If you get an error there, try with minimal
But adter 8.5.1, always use the GZ filedirectly. Do not use the BIN, do not uncompress the GZ (Tasmota will do it in memory while flashing).

> Why did the flashing "brick" the device when flashing from minimal to full bin

What is bricking the device is performing the upgrade path using only minimal like minimal 7.x => minimal 8.5.1 => minimal 9.1 ...
Because minimal is not meant to support the upgrade process.
When using minimal, you should immediatly flash the "lite" or "full" of the newt release
For exemple : full 7.x => minimal 8.5.1 => full 8.5.1 => minimal 9.1 = full 9.1 ... would work
But it's still more complicated and can lead to error 
The simplest being full 7.x => tasmota-lite.bin 7.2 => tasmota-lite.bin 8.5.1 => tasmota-lite.bin.gz 9.1 => tasmota.bin.gz 10.1.0

> Is there no estimated flash size check implemented in this release 

All standard tasmota builds are meant for devices with 1MB. Check is done on that purpose
What minimal is missig is not the flash size check but the process to upgrade the settings.
Lite can do that.

> that no OTA update is possible for the Sonoff T1,

Why ? I don't see any reason for the T1 to be different.

Hope this helps

**Reply by trex2000** [reply]
Thanks, I've tried on another device with your suggestions and it worked. Thanks alot for the explanation!
